fitbenchmarking/results_processing/performance_profiler.py

```diff
@@ -360,15 +409,26 @@ def __init__(self, profile_name, pp_df, group_label):
         self.data = pp_df
         self.profile_name = profile_name
         self.group_label = group_label
-        self.identif = self.group_label + '-' + self.profile_name
+        self.id = self.group_label + '-' + self.profile_name
 
         self.default_opt = []
-        for solver in self.data['solver'].unique():
+        for solver in self.data.columns:
             self.default_opt.append({
                 "value": solver,
                 "label": solver
             })
 
+        self.current_styles = {}
+        self.avail_styles = {}
+
+        self.colors = ['#1f77b4', '#ff7f0e', '#2ca02c', '#d62728', '#9467bd',
+                       '#8c564b', '#e377c2', '#7f7f7f', '#bcbd22', '#17becf']
+        self.linestyles = ['solid', 'dash', 'dashdot']
+
+        for comb in list(itertools.product(self.linestyles, self.colors)):
+            comb_id = comb[0]+comb[1]
+            self.avail_styles[comb_id] = comb
```

Comment:
I'm not sure I see the benefit of this being a dict, now that I reread it. I'll put a few suggestions in and hopefully you can see what I mean.
```suggestion
        self.avail_styles = list(itertools.product(self.linestyles, self.colors))
```